### PR TITLE
doc: describe Hashicorp Vault with a ServiceAccount per Tenant

### LINKED ISSUE #2222: New KMS that supports Hashicorp Vault with a ServiceAccount per Tenant
Introduce a new KMS implementation that has support for Hashicorp Vault where each Tenant can configure access with its own ServiceAccount.

- [x] design document (#2227)
- [x] implementation (#2246)
- [ ] user documentation

### PR DESCRIPTION
**Pull request by nixpanic**
In addition to the single ServiceAccount KMS support for Hashicorp
Vault, Ceph-CSI can now use a ServiceAccount per Tenant as well. This
adds the user-documentation with references to the example deployment
files.

Closes: #2222

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

File: docs/deploy-rbd.md [hunk 1]
```diff
@@ -299,7 +299,7 @@ it in a Kubernetes Secret (by default) called `ceph-csi-kms-token`, where the
 Vault Token is stored in the `token` key as shown in [the
 example](../examples/kms/vault/tenant-token.yaml).
 
-#### Configuring HashiCorp Vault with Kubernetes ServiceAccount
+#### Configuring HashiCorp Vault with a single Kubernetes ServiceAccount
```
Comment (pkalever) [inline docs/deploy-rbd.md:302]: nit: s/single/Single/ ?

Comment (nixpanic) [reply docs/deploy-rbd.md:302]: "Single" is not a name of a component/project/object, or start of the title, so I kept it lowecase

Comment (pkalever) [reply docs/deploy-rbd.md:302]: This is a title case, so it should be **Single** :-)

**Review by yati1998 — COMMENTED** [review]
LGTM

File: docs/deploy-rbd.md (hunk 1 above)
Comment (pkalever) [reply docs/deploy-rbd.md:302]: Ah! please ignore, I see now that nowhere the doc is following the title case.

**Review by pkalever — APPROVED** [review]
LGTM

**Review by Madhu-1 — DISMISSED** [review]

**Review by humblec — DISMISSED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

@humblec it seemt Mergify dropped the approvals again, could you re-add yours and maybe add the `[ready-to-merge]` label?

**Review by humblec — APPROVED** [review]

**Pull request merged**